src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -260,14 +266,32 @@ String actualApiUri() {
     return Util.fixEmpty(apiUri) == null ? "https://api.github.com" : apiUri;
   }
 
-  private AppInstallationToken getToken(GitHub gitHub) {
+  private AppInstallationToken getToken(final GitHub gitHub) {
+    return getToken(gitHub, owner);
+  }
+
+  private AppInstallationToken getToken(final GitHub gitHub, final String owner) {
+    AppInstallationToken cachedToken = null;
     synchronized (this) {
       try {
+        if (cachedTokens == null) {
+          cachedTokens = new ArrayList<>();
+        } else {
+          Optional<AppInstallationToken> tempToken =
+              cachedTokens.stream()
```

Comment:
if you are just going to keep one token wouldn't a Map make more sense and be more clear?

Reply:
> I tried Map and changed it to List, I think I did that to avoid adding overrides to serialize/deserialize as this is transient field.

Not sure why this is a reason to use List instead of Map.  

> I think it is good to have that owner field with actual class just in case if we are going to use this token for future use cases across other classes.

I'm not sure what you mean here.